test/test_config.py

```diff
@@ -1,17 +1,57 @@
 import os.path
 import tempfile
-import unittest
+from unittest import TestCase
 from pathlib import Path
-
 from config import AlltalkConfig, AlltalkTTSEnginesConfig, AlltalkNewEnginesConfig
 
 
-class TestAlltalkConfig(unittest.TestCase):
+class TestAlltalkConfig(TestCase):
 
     def setUp(self):
         self.config = AlltalkConfig.get_instance()
         self.config.reload()
 
+    def test_default_values_loaded(self):
+        cfg = AlltalkConfig(Path(__file__).parent.resolve() / 'empty.json')
+
+        # Since the default config file is expected to also contain the
+        # default values, we can simply check that both dictionaries are identical:
+        self.assertEqual(self.config.to_dict(), cfg.to_dict())
+
+    def test_no_default_values_missing(self):
+        # Loading the empty JSON will populate the config with defaults from the code:
+        cfg = AlltalkConfig(Path(__file__).parent.resolve() / 'empty.json')
+        with tempfile.NamedTemporaryFile(suffix=".json") as tmp:
+            cfg.save(tmp.name)
+            # Compare the defaults from code (when written to a file) to the actual default config file:
+            with open(tmp.name, "r") as file1:
+                with open(self.config.get_config_path(), "r") as file2:
+                    self.assertEqual(file1.read(), file2.read())
+
+    def test_values_merged_with_defaults(self):
```

Comment:
This is a test for the complex issue of merging partial config json files with the defaults defined in code. That's what pydantic is actually good for to handle automatically.